authenticate/handlers.go

```diff
@@ -226,18 +226,34 @@ func (a *Authenticate) SignOut(w http.ResponseWriter, r *http.Request) error {
 		return httputil.NewError(http.StatusBadRequest, err)
 	}
 
+	redirectURL, err := urlutil.ParseAndValidateURL(r.FormValue(urlutil.QueryRedirectURI))
+	if err != nil {
+		return httputil.NewError(http.StatusBadRequest, err)
+	}
+
 	a.sessionStore.ClearSession(w, r)
 	err = a.provider.Revoke(r.Context(), s.AccessToken)
 	if errors.Is(err, identity.ErrRevokeNotImplemented) {
 		log.FromRequest(r).Warn().Err(err).Msg("authenticate: revoke not implemented")
+	} else if errors.Is(err, identity.ErrNoRevokeWithEndSessionURL) {
```

Comment:
The control flow is rather awkward here.

Reply:
Thanks for the fix in the PR, it was helpful.